modules/social_features/social_user/social_user.module

```diff
@@ -675,31 +675,40 @@ function social_user_ensure_help_text_fieldset(&$form) {
  * displays the one with the lowest weight (highest priority).
  */
 function social_user_user_format_name_alter(&$name, AccountInterface $account) {
-  // We always add the username as fallback suggestion.
-  $suggestions = [
-    'username' => [
-      'name' => $account->getAccountName(),
-      'weight' => PHP_INT_MAX,
-    ],
-  ];
+  $suggested_names = &drupal_static(__FUNCTION__, []);
+
+  if (empty($suggested_names[$account->id()])) {
+    // We always add the username as fallback suggestion.
+    $suggestions = [
+      'username' => [
+        'name' => $account->getAccountName(),
+        'weight' => PHP_INT_MAX,
+      ],
+    ];
 
-  $suggestions += \Drupal::moduleHandler()->invokeAll('social_user_name_display_suggestions', [$account]);
-  \Drupal::moduleHandler()->alter('social_user_name_display_suggestions', $suggestions, $account);
+    $suggestions += \Drupal::moduleHandler()->invokeAll('social_user_name_display_suggestions', [$account]);
+    \Drupal::moduleHandler()->alter('social_user_name_display_suggestions', $suggestions, $account);
 
-  // We could use -PHP_INT_MAX but someone being clever might shoot themselves.
-  $lowest_weight = NULL;
+    // We could use -PHP_INT_MAX but someone being clever might shoot themselves.
+    $lowest_weight = NULL;
 
-  // If our array ends up being empty then we just stick to Drupal's default.
-  foreach ($suggestions as $suggestion) {
-    // Allow weight to be omitted and default to 0.
-    $suggestion['weight'] = $suggestion['weight'] ?? 0;
+    // If our array ends up being empty then we just stick to Drupal's default.
+    foreach ($suggestions as $suggestion) {
+      // Allow weight to be omitted and default to 0.
+      $suggestion['weight'] = $suggestion['weight'] ?? 0;
 
-    // If the suggestion's weight is equal to a previous entry then we use the
-    // item that appears first in the array.
-    if ($lowest_weight === NULL || $suggestion['weight'] < $lowest_weight) {
-      $name = $suggestion['name'];
-      $lowest_weight = $suggestion['weight'];
+      // If the suggestion's weight is equal to a previous entry then we use the
+      // item that appears first in the array.
+      if ($lowest_weight === NULL || $suggestion['weight'] < $lowest_weight) {
+        $suggested_names[$account->id()] = $suggestion['name'];
+        $lowest_weight = $suggestion['weight'];
+      }
     }
+
+    $name = $suggested_names[$account->id()];
+  }
+  else {
+    $name = $suggested_names[$account->id()];
```

Comment:
We can move the name assignment out of the if-statement and remove the else. This makes the code slightly easier to comprehend and follows the more common structure of 
```
$cache
if (not calculated) { calculate and fill }
use cache
```

```suggestion
  }

  $name = $suggested_names[$account->id()];
```